app/Http/Controllers/Auth/RegisterController.php

```diff
@@ -68,6 +71,36 @@ protected function create(array $data)
             'last_name' => $data['last_name'],
             'email' => $data['email'],
             'password' => bcrypt($data['password']),
+            'email_token' => base64_encode($data['email'])
         ]);
     }
+
+    /**
+     * Handle a registration request for the application.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function register(Request $request)
+    {
+        $this->validator($request->all())->validate();
+        event(new Registered($user = $this->create($request->all())));
+        dispatch(new SendVerificationEmail($user));
+        return view('email.verification');
+    }
+
+    /**
+     * Handle a registration request for the application.
+     *
+     * @param $token
+     * @return \Illuminate\Http\Response
+     */
+    public function verify($token)
+    {
+        $user = User::where('email_token',$token)->first();
```

Comment:
Always add a comma + space between arguments: `User::where('email_token, $token)->first();`

Reply:
You also have to take care of what happens when the user cannot be found. Maybe the link is broken for some reason. In that case, `$user` will be `null`. Best to handle those situations as well and return a nice error message to the user.